regEx/challenges-04.test.js

```diff
@@ -0,0 +1,240 @@
+// regular expression pattern to match all words that begin with a capital letter.
+//
+// Return an array containing all the matches.
+// ------------------------------------------------------------------------------------------------
+
+const isCapitalized = (string) => {
+  let regex = /^[A-Z]/;
+  const capitalArr = [];
+  let newStr = string.split(' ');
+  newStr.forEach( (word) => {
+    if(regex.test(word)){
+      capitalArr.push(word);
+    }
+  });
+  return capitalArr;
+};
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 3
+//
+// Write a function named citiesAtoJ that takes in an array of city names and uses a regular expression
+// pattern to return a new array containing any cities that begin with the letters A through J, inclusive.
+// ------------------------------------------------------------------------------------------------
+
+const citiesAtoJ = (cities) => {
+  let regex = /^[A-J]/;
+  const newArr = [];
+  cities.forEach( (city) => {
+    if(regex.test(city)){
+      newArr.push(city);
+    }
+  });
+  return newArr;
+};
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 4
+//
+// You have created a game application and begin by asking users an easy question: In which month is Halloween?
+//
+// Write a function named matchMonth which uses a regular expression pattern to match any of these inputs:
+// October, Oct, october, oct
+//
+// If the user enters any of these four inputs, return true. For ANY other input, return false.
+// Do not use the vertical bar (pipe) in your pattern.
+// ------------------------------------------------------------------------------------------------
+
+const matchMonth = (input) => {
+  let regex = /^([Oo]ct)(ober)?$/;
+  return regex.test(input);
+};
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 5
+//
+// Write a function named noPunctuation that contains a regular expression pattern to find
+// all of the words that contain a space immediately at the end of the word. Return an array of all
+// such words, still containing the space at the end.
+//
+// For example, if given the string "Hello, and have a wonderful day!",
+// the word "Hello, " would not be returned because it is immediately followed by a comma,
+// and the word "day!" would not be returned because it is immediately followed by an exclamation point.
+// The expected output is ["and ", "have ", "a ", "wonderful "].
+// ------------------------------------------------------------------------------------------------
+
+const noPunctuation = input => {
+  let regex = /[a-z]$/;
+  let newArr = input.split(' ');
+  const answer = [];
+  newArr.forEach( (word) => {
+    if(regex.test(word)){
+      answer.push(word+' ');
+    }
+  });
+  return answer;
+};
```

Comment:
you should be able to do this using .match() and not splitting your input and using this 'hack'. This would also cut your code down to 2 lines.